File: frontend/src/app/models/planning.models.ts
export interface ComponentGapDocument {
  componentPartId: string;
  partNumber: string;
  componentName: string;
  partType: string;
  workCenterId: string | null;
  workCenterName: string | null;
  quantityPer: number;
  targetQuantity: number;
  quantityRequired: number;
  quantityOnHand: number;
  quantityAllocated: number;
  quantityOnOrder: number;
  quantityAvailable: number;
  shortage: number;
  standardBuildDays: number;
  standardLeadDays: number;
  projectedReadyDays: number;
}

File: api/Models/Planning/PlanningModels.cs
namespace Naologic_API.Models.Planning;

public sealed record ComponentGapDocument(
    string ComponentPartId,
    string PartNumber,
    string ComponentName,
    string PartType,
    string? WorkCenterId,
    string? WorkCenterName,
    decimal QuantityPer,
    decimal TargetQuantity,
    decimal QuantityRequired,
    decimal QuantityOnHand,
    decimal QuantityAllocated,
    decimal QuantityOnOrder,
    decimal QuantityAvailable,
    decimal Shortage,
    int StandardBuildDays,
    int StandardLeadDays,
    int ProjectedReadyDays);
